File: src/main/java/indi/midreamsheep/app/markdown/context/config/AbstractStandardConfig.java
package indi.midreamsheep.app.markdown.context.config;

import cn.hutool.json.JSONUtil;
import indi.midreamsheep.app.markdown.config.AbstractConfig;
import indi.midreamsheep.app.markdown.context.di.init.annotation.Init;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

public class AbstractStandardConfig extends AbstractConfig {

    private static final String rootPath = System.getProperty("user.dir");


    @Init
    public void set(){
        init();
    }

    @Override
    public void write() {
        File file = new File(rootPath + "/configs/standard/" + geiFileName() + ".json");
        try {
            if (!file.getParentFile().exists()) {
                file.getParentFile().mkdirs();
            }
            file.createNewFile();
            new FileOutputStream(file).write(JSONUtil.toJsonStr(this).getBytes());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public String getJsonString() {
        File file = new File(rootPath + "/configs/standard/" + geiFileName() + ".json");
        if (file.exists()) {
            try {
                byte[] bytes = new FileInputStream(file).readAllBytes();
                return new String(bytes);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }
        try {
            if (!file.getParentFile().exists()) {
                file.getParentFile().mkdirs();
            }
            file.createNewFile();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        return "{}";
    }

    @Override
    public String getConfigName() {
        return this.getClass().getSimpleName();
    }

    protected String geiFileName(){
        return this.getClass().getSimpleName();
    }
}

File: src/main/kotlin/indi/midreamsheep/app/markdown/setting/AbstractStandardConfig.kt
package indi.midreamsheep.app.markdown.setting

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import indi.midreamsheep.app.markdown.context.di.init.annotation.Init
import indi.midreamsheep.app.markdown.context.setting.tool.SettingUtil
import indi.midreamsheep.app.markdown.ui.setting.ExpandableItem
import java.io.File

abstract class AbstractStandardConfig:AbstractConfig {
    object Constants{
       @JvmField  val rootPath = System.getProperty("user.dir")+ File.separator + "configs" + File.separator
    }

    @Init
    fun init(){
        val file = File(Constants.rootPath+getConfigName()+".json")
        if (!file.exists()){
            return
        }
        val readText = file.readText()
        SettingUtil.setData(this,readText)
    }

    override fun getComposable(): @Composable () -> Unit {
        val list = mutableListOf<@Composable () -> Unit>()
        for (config in SettingUtil.getConfigs(this)) {
            list.add(config.getComposable())
        }
        return {
            ExpandableItem(
                title = getConfigName(),
                subItem = {
                    for (composable in list) {
                        composable()
                    }
                },
                modifier = Modifier.fillMaxWidth().padding(bottom = 5.dp).clip(RoundedCornerShape(3.dp)).background(Color.White)
            )
        }
    }

    override fun getConfigName(): String {
        return this.javaClass.simpleName
    }

    override fun write(): Pair<Boolean, String> {
        //写入文件
        val file = File(Constants.rootPath+getConfigName()+".json")
        if (!file.exists()){
            file.createNewFile()
        }
        file.writeText(getJsonString())
        return Pair(true,"")
    }

    private fun getJsonString():String{
        return SettingUtil.getJson(this)
    }
}